### PR TITLE
[POAE7-2721] IN expr support for primitive type in Nextgen

### PR DESCRIPTION
**Pull request by yma11**
### What changes were proposed in this pull request?
 IN expr support for primitive type in Nextgen

### Why are the changes needed?
in support

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
UT

### CODE REVIEW

**Comment by yma11** [comment]
@jikunshang @bigPYJ1151 @kd97 please help take a review. Thanks.

File: src/cider/type/plan/BinaryExpr.cpp [hunk 1]
```diff
@@ -52,8 +53,10 @@ JITExprValue& BinOper::codegen(JITFunction& func) {
     // string binops, should only be comparisons
     const auto optype = get_optype();
     if (IS_COMPARISON(optype)) {
-      VarSizeJITExprValue lhs_val(lhs->codegen(func));
-      VarSizeJITExprValue rhs_val(rhs->codegen(func));
+      std::cout << lhs->toString() << std::endl;
```
Comment (haiweizh) [inline src/cider/type/plan/BinaryExpr.cpp:56]: Remove this cout.

File: src/cider/exec/nextgen/function/CiderSetFunctions.cpp [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    auto cider_set =                                                              \
+        reinterpret_cast<cider::exec::nextgen::context::CiderInt64Set*>(set_ptr); \
+    return cider_set->contains(val);                                              \
+  }
+
+DEF_CIDER_INT64_SET_CONTAINS(8)
+DEF_CIDER_INT64_SET_CONTAINS(16)
+DEF_CIDER_INT64_SET_CONTAINS(32)
+DEF_CIDER_INT64_SET_CONTAINS(64)
+
```
Comment (kd97) [inline src/cider/exec/nextgen/function/CiderSetFunctions.cpp:36]: better to define and gen:
`cider_set_contains_##type##_val(int8_t* set_ptr, const type val)`
 `DEF_CIDER_INT64_SET_CONTAINS(int8_t)` ?

File: src/cider/function/CMakeLists.txt [hunk 3]
```diff
@@ -70,6 +70,7 @@ add_custom_command(
     ${CMAKE_CURRENT_SOURCE_DIR}/../exec/nextgen/context/ContextRuntimeFunctions.h
     ${CMAKE_CURRENT_SOURCE_DIR}/../exec/nextgen/operators/OperatorRuntimeFunctions.h
     ${CMAKE_CURRENT_SOURCE_DIR}/../exec/nextgen/function/CiderStringFunctions.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../exec/nextgen/function/CiderSetFunctions.cpp
   OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/RuntimeFunctions.bc
   COMMAND
```
Comment (kd97) [inline src/cider/function/CMakeLists.txt:75]: can traverse and add all files in function folder automatically?

File: src/cider/exec/nextgen/context/CiderSet.h [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+    CIDER_THROW(CiderRuntimeException,
+                "CiderDoubleSet doesn't support insert int32_t value.");
+  }
+
+  void insert(int64_t key_val) {
+    CIDER_THROW(CiderRuntimeException,
+                "CiderDoubleSet doesn't support insert int64_t value.");
+  }
+
+  void insert(float key_val) { set_.insert(key_val); }
```
[the remark this answers is not in the document]
Comment (yma11) [reply src/cider/exec/nextgen/context/CiderSet.h]: Both float and double has precision issue as offline tested. Current float/double IN test are verified by UT but can't guarantee there is no problem in real E2E. Will keep track on this.

File: src/cider/type/plan/InValues.cpp [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+  auto& in_val_ti = val_list.front()->get_type_info();
+  for (auto in_val : val_list) {
+    const auto in_val_const =
+        dynamic_cast<const Analyzer::Constant*>(extract_cast_arg(in_val.get()));
+    if (!in_val_const) {
+      CIDER_THROW(CiderCompileException, "InValues only support constant value list.");
+    }
+    switch (in_val_ti.get_type()) {
+      case kTINYINT: {
+        // For case like IN [1, 2, NULL], NULL will be ignored
```
[the remark this answers is not in the document]
Comment (yma11) [reply src/cider/type/plan/InValues.cpp]: Done.

File: src/cider/type/plan/BinaryExpr.cpp (hunk 1 above)
Comment (yma11) [reply src/cider/type/plan/BinaryExpr.cpp]: Removed.

File: src/cider/exec/nextgen/function/CiderSetFunctions.cpp (hunk 2 above)
Comment (yma11) [reply src/cider/exec/nextgen/function/CiderSetFunctions.cpp:36]: Updated.

File: src/cider/function/CMakeLists.txt (hunk 3 above)
Comment (yma11) [reply src/cider/function/CMakeLists.txt:75]: Current the function files exist in different folders and not quite good to use `foreach` to apply all files. We can consolidate these files after fully migrated to nextgen.

**Comment by yma11** [comment]
@jikunshang @kd97 @haiweizh @bigPYJ1151 Please help review again. Thanks.

**Review by kd97 — APPROVED** [review]

File: src/cider/exec/nextgen/context/CiderSet.h [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+  virtual void insert(double key_val) = 0;
+  virtual void insert(std::string key_val) = 0;
+
+  virtual bool contains(int8_t key_val) = 0;
+  virtual bool contains(int16_t key_val) = 0;
+  virtual bool contains(int32_t key_val) = 0;
+  virtual bool contains(int64_t key_val) = 0;
+  virtual bool contains(float key_val) = 0;
+  virtual bool contains(double key_val) = 0;
+  virtual bool contains(std::string key_val) = 0;
```
Comment (bigPYJ1151) [inline src/cider/exec/nextgen/context/CiderSet.h:48]: Better to pass by reference?

**Review by bigPYJ1151 — APPROVED** [review]

**Pull request merged**